File: frontend-react/src/services/orderService.ts
import api, { ENDPOINTS } from '../config/api';
import type { CustomerOrder, OrderItem, OrderStatus, PageResponse } from '../types';

export const orderService = {
  getAll: async (page = 0, size = 10): Promise<PageResponse<CustomerOrder>> => {
    const response = await api.get(ENDPOINTS.ORDERS, { params: { page, size, sort: 'id,desc' } });
    return response.data;
  },

  getById: async (id: number): Promise<CustomerOrder> => {
    const response = await api.get(`${ENDPOINTS.ORDERS}/${id}`);
    return response.data;
  },

  getByStatus: async (status: OrderStatus, page = 0, size = 10): Promise<PageResponse<CustomerOrder>> => {
    const response = await api.get(ENDPOINTS.ORDERS, { params: { status, page, size } });
    return response.data;
  },

  create: async (order: Partial<CustomerOrder>): Promise<CustomerOrder> => {
    const response = await api.post(ENDPOINTS.ORDERS, order);
    return response.data;
  },

  updateStatus: async (id: number, status: OrderStatus): Promise<CustomerOrder> => {
    const response = await api.patch(`${ENDPOINTS.ORDERS}/${id}/status`, null, { params: { status } });
    return response.data;
  },

  createItem: async (item: Partial<OrderItem>): Promise<OrderItem> => {
    const response = await api.post(ENDPOINTS.ORDER_ITEMS, item);
    return response.data;
  },

  getItemsByOrder: async (orderId: number): Promise<OrderItem[]> => {
    const response = await api.get(ENDPOINTS.ORDER_ITEMS, { params: { orderId } });
    return response.data.content;
  },
};

File: frontend-flutter/lib/services/order_service.dart
import '../config/api_config.dart';
import '../models/models.dart';
import 'api_service.dart';

class OrderService {
  final _api = ApiService();

  Future<List<CustomerOrder>> getAll({int page = 0, int size = 10}) async {
    final response = await _api.dio.get(
      ApiConfig.ordersEndpoint,
      queryParameters: {'page': page, 'size': size, 'sort': 'id,desc'},
    );
    return (response.data['content'] as List)
        .map((json) => CustomerOrder.fromJson(json))
        .toList();
  }

  Future<CustomerOrder> getById(int id) async {
    final response = await _api.dio.get('${ApiConfig.ordersEndpoint}/$id');
    return CustomerOrder.fromJson(response.data);
  }

  Future<CustomerOrder> create(Map<String, dynamic> data) async {
    final response = await _api.dio.post(ApiConfig.ordersEndpoint, data: data);
    return CustomerOrder.fromJson(response.data);
  }

  Future<OrderItem> createItem(Map<String, dynamic> data) async {
    final response = await _api.dio.post(ApiConfig.orderItemsEndpoint, data: data);
    return OrderItem.fromJson(response.data);
  }
}
